main.go

```diff
@@ -80,21 +81,43 @@ func main() {
 	var bindPort int
 	var serveAddress string
 	var dataSource string
+	var configFile string
+	var err error
 
 	flag.StringVar(&bindIp, "b", "127.0.0.1", "IP address for the server to listen on")
 	flag.IntVar(&bindPort, "p", 9000, "port for the server to listen on")
 	flag.StringVar(&serveAddress, "s", "", "IP:Port that connections will see the server at (defaults to bind address)")
 	flag.StringVar(&dataSource, "d", "", "data source (path to .gpkg file)")
+	flag.StringVar(&configFile, "c", "", "config (path to .toml file)")
 
 	flag.Parse()
 
+	// Configuration logic
+	// 1. config.Configuration gets set at startup (via config.init())
+	// 2. if -c is passed, config file overrides
+	// 3. if other command line arguments are passed, they override previous settings
+
```

Comment:
`// 4. If no  data provider is supplied by any of these means, the working directory is scanned for .gpkg files, then the 'data/' and 'test_data/' directories.`